Update torch-xpu-ops commit pin

toyxu opened the pull request:
Update the torch-xpu-ops commit to [655fa9bc7f88ab5bd3766b5f2fd5b43989c2caca](https://github.com/intel/torch-xpu-ops/commit/655fa9bc7f88ab5bd3766b5f2fd5b43989c2caca), including:
- Update commit pin to xpu-ops main branch
- Fixes batch_norm numeric error by adding additional boundary check
- Enable two operators: fft & jagged_to_padded_dense
- XCCL relevant changes:
1. Cache `cclStream` to improve performance.
2. Add support for complex datatypes in `allgather` and `broadcast`.
3. Support `coalescing` operations and `batch_isend_irecv`.
4. Introduce additional logging; use `export TORCH_CPP_LOG_LEVEL=INFO`.


cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jerryzh168 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

toyxu committed:
Update xpu.txt

xytintel commented:
The CI issue will be fixed by https://github.com/pytorch/pytorch/pull/149862

guangyey commented on third_party/xpu.txt:
```diff
@@ -1 +1 @@
-98c808dea6de7330c415aa777d6921944cf79887
+7bb07d719a839bcdb3734e2802a6364da29a521e
```
```suggestion
bd0176dd354b2953e14c908fd1a323e90a1724a0
```

guangyey committed:
Update third_party/xpu.txt

EikanWang commented:
@xytintel , For the Windows failure, it is not a bug fix because the Windows failure is an intermediate step of the development of the torch-xpu-ops repo, and the failure is not exposed to the torch.

EikanWang commented:
@xytintel , could you help refine the description a little bit to make it more accurate?

toyxu committed:
Update xpu.txt

xytintel commented:
`NotImplementedError: The operator 'aten::_conv_depthwise2d' is not currently implemented for the XPU device` https://github.com/pytorch/pytorch/issues/151308

toyxu committed:
Update xpu.txt

EikanWang commented:
@pytorchbot merge -i

EikanWang commented:
@pytorchbot merge -i

guangyey commented:
@pytorchbot merge -i

guangyey commented:
@pytorchbot merge -f "XPU pin commit update, all XPU-related tests finished, lint is green"

etaf commented:
Sorry to revert  this PR, it cause regression in Inductor UT #152020
@pytorchbot revert

etaf commented:
@pytorchbot revert -m "Inductor UT regression" -c "ignoredsignal"

guangyey commented:
@pytorchbot rebase

swolchok committed:
irangeify ReducedPrecisionFloatGemvKernel.cpp (#152232)

We should be using irange, especially because we had 32-bit overflow issues in this file recently.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/152232
Approved by: https://github.com/Skylion007

swolchok committed:
ReducedPrecisionFloatGemvFastPathKernel: Correctly type parallel_for lambda arguments as int64_t (#152233)

This plus the previous irangeification PR seem like a better fix for #150637 than #150949 to me -- should make sure we are using 64-bit math for indexing everywhere.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/152233
Approved by: https://github.com/Skylion007, https://github.com/cyyever
ghstack dependencies: #152232

anmyachev committed:
[Inductor UT] Generalize device-bias code in `test_flex_attention.py` (#151937)

@EikanWang @etaf @guangyey please take a look

Pull Request resolved: https://github.com/pytorch/pytorch/pull/151937
Approved by: https://github.com/liangan1, https://github.com/drisspg

aartbik committed:
Add check for 2-dim mask to COO mask computation (#151940)

Follow up on discussion on https://github.com/pytorch/pytorch/pull/151794 Related to all fixes for https://github.com/pytorch/pytorch/issues/151351

Pull Request resolved: https://github.com/pytorch/pytorch/pull/151940
Approved by: https://github.com/Skylion007

Isalia20 committed:
[MPS] col2im kernel implementation (#152282)

Fixes #151820
Also requested in #141287

Mainly based on the cuda kernel implementations

Pull Request resolved: https://github.com/pytorch/pytorch/pull/152282
Approved by: https://github.com/malfet

malfet committed:
[BE] Migrate dtype_abbrs into one location (#152229)

Namely `torch.utils._dtype_abbrs.dtype_abbrs`

Before that it was defined in various forms of completeness in
https://github.com/pytorch/pytorch/blob/c02edba86388d1f86a78bce99d16c5405b54086e/torch/fx/graph.py#L215,
https://github.com/pytorch/pytorch/blob/c02edba86388d1f86a78bce99d16c5405b54086e/torch/testing/_internal/common_utils.py#L5226
 and https://github.com/pytorch/pytorch/blob/c02edba86388d1f86a78bce99d16c5405b54086e/torch/testing/_internal/logging_tensor.py#L17

TODO:
 - Add linter that `torch.testing._internal` module is not referenced from any of the public facing APIs, as it can have extra dependencies such as `expect_test`

Fixes https://github.com/pytorch/pytorch/issues/152225

Pull Request resolved: https://github.com/pytorch/pytorch/pull/152229
Approved by: https://github.com/clee2000, https://github.com/Skylion007

exclamaforte committed:
Enable max autotune for AOTInductor benchmark (#149309)

With this PR, AOTinductor can choose to run into max-autotune mode when benchmarking.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/149309
Approved by: https://github.com/desertfire

Co-authored-by: Gabriel Ferns <gabeferns@meta.com>

toyxu committed:
Update xpu.txt

guangyey commented:
@pytorchbot rebase
